### ISSUE TITLE
node-pre-gyp error 404 when installing cli tools on FreeBSD 12.2

### ISSUE THREAD

**Issue by ctag** [Chocobozzz/PeerTube#4391]
**Describe the current behavior**
When installing the CLI tools (following directions from https://docs.joinpeertube.org/maintain-tools) I get to the line `npm run setup:cli` and it returns an error.

**Steps to reproduce:**

1. Clone a new copy of PeerTube onto the peertube server.
2. cd into cloned dir
3. Switch to bash shell (for variables to work)
4. Run `NOCLIENT=1 yarn install --pure-lockfile`
5. Run `npm run setup:cli`

**Describe the expected behavior**

**Additional information**

Error text:
```
# npm run setup:cli

> peertube@3.4.0 setup:cli /var/www/peertube/cli-tools/PeerTube
> bash ./scripts/setup/cli.sh

yarn install v1.22.10
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
$ test -n "$NOCLIENT" || (cd client && yarn install --pure-lockfile)
Done in 1.23s.
yarn install v1.22.10
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning "wrtc@0.4.7" is missing a bundled dependency "node-pre-gyp". This should be reported to the package maintainer.
[4/4] Building fresh packages...
[1/5] ⢀ bufferutil
[2/5] ⢀ utf-8-validate
[3/5] ⢀ utp-native
[-/5] ⢀ waiting...
error /var/www/peertube/cli-tools/PeerTube/server/tools/node_modules/wrtc: Command failed.
Exit code: 1
Command: node scripts/download-prebuilt.js
Arguments: 
Directory: /var/www/peertube/cli-tools/PeerTube/server/tools/node_modules/wrtc
Output:
node-pre-gyp WARN Using needle for node-pre-gyp https download 
node-pre-gyp ERR! install error 
node-pre-gyp ERR! stack Error: 404 status code downloading tarball https://node-webrtc.s3.amazonaws.com/wrtc/v0.4.7/Release/freebsd-x64.tar.gz
node-pre-gyp ERR! stack     at PassThrough.<anonymous> (/var/www/peertube/cli-tools/PeerTube/server/tools/node_modules/node-pre-gyp/lib/install.js:149:27)
node-pre-gyp ERR! stack     at PassThrough.emit (events.js:327:22)
node-pre-gyp ERR! stack     at ClientRequest.<anonymous> (/var/www/peertube/cli-tools/PeerTube/server/tools/node_modules/needle/lib/needle.js:508:9)
node-pre-gyp ERR! stack     at Object.onceWrapper (events.js:422:26)
node-pre-gyp ERR! stack     at ClientRequest.emit (events.js:315:20)
node-pre-gyp ERR! stack     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:641:27)
node-pre-gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:126:17)
node-pre-gyp ERR! stack     at TLSSocket.socketOnData (_http_client.js:509:22)
node-pre-gyp ERR! stack     at TLSSocket.emit (events.js:315:20)
node-pre-gyp ERR! stack     at addChunk (internal/streams/readable.js:309:12)
node-pre-gyp ERR! System FreeBSD 12.2-RELEASE-p9
node-pre-gyp ERR! command "/usr/local/bin/node" "/var/www/peertube/cli-tools/PeerTube/server/tools/node_modules/wrtc/node_modules/.bin/node-pre-gyp" "install"
node-pre-gyp ERR! cwd /var/www/peertube/cli-tools/PeerTube/server/tools/node_modules/wrtc
node-pre-gyp ERR! node -v v14.16.0




npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! peertube@3.4.0 setup:cli: `bash ./scripts/setup/cli.sh`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the peertube@3.4.0 setup:cli script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-09-09T17_52_22_894Z-debug.log
```

Following that link (https://node-webrtc.s3.amazonaws.com/wrtc/v0.4.7/Release/freebsd-x64.tar.gz) returns NoSuchKey. It looks to a layperson like me that this package isn't available for FreeBSD?

* PeerTube instance:
  * URL: video.berocs.com
  * version: 3.4.0
  * NodeJS version: node14
  * Ffmpeg version: 4.3.2

* Browser name, version and platforms on which you could reproduce the bug: 
* Link to browser console log if relevant:
* Link to server log if relevant (`journalctl` or `/var/www/peertube/storage/logs/`):

**Comment by ctag** [comment]
I'd like to mention here for anyone as clueless as myself that you do not need to execute the CLI tools **on the Peertube webserver**. In my case I was able to build and run them on an Ubuntu machine and it worked well.

I'd be glad to either close this issue or keep it open and run any testing/troubleshooting steps on my FreeBSD machine so that the CLI tools will work for others.

**Comment by Aesthetikx** [comment]
I was able to build and run the CLI tools by doing the following:

1) Removing `webtorrent-hybrid` from `server/tools/package.json`, as this depends on `wrtc` which doesn't work on FreeBSD (for a myriad of reasons, the rabbit hole is deep). This doesn't seem to prevent `npm run setup:cli` from working.
2) Removing any reference to `application-config` and just returning `defaultSettings` in `getSettings()` in `server/tools/cli.ts`. That package also doesn't work with FreeBSD, and the package that would need to (trivially) change has not been updated in 6 years, so perhaps the dependency needs to go at some point haha.

At this point I was able to run `peertube-import-videos.js` successfully.
